src/shogun/lib/observers/ParameterObserver.cpp

```diff
@@ -33,37 +33,39 @@
 *
 */
 #include <shogun/lib/RefCount.h>
-#include <shogun/lib/observers/ParameterObserverInterface.h>
+#include <shogun/lib/observers/ParameterObserver.h>
+#include <shogun/base/Parameter.h>
 
 using namespace shogun;
 
-ParameterObserverInterface::ParameterObserverInterface() : m_parameters()
+ParameterObserver::ParameterObserver() : m_observed_parameters(), m_subscription_id(-1)
 {
+	SG_ADD(&m_subscription_id, "subscription_id", "Id of the subscription to an object.");
```

Comment:
-> "id" :D

Reply:
same as above for the `trained_machine` stuff. I imagine the user asking himself "id of what?" when it is trying to get it.